app/services/hyrax/listeners/metadata_index_listener.rb

```diff
@@ -16,7 +16,17 @@ class MetadataIndexListener
       #
       # @param event [Dry::Event]
       def on_object_metadata_updated(event)
-        Hyrax.index_adapter.save(resource: event[:object])
+        Hyrax.index_adapter.save(resource:
+          Hyrax::ValkyrieIndexer.for(resource: event[:object]).resource)
+      end
+
+      ##
+      # Index the resource.
+      #
+      # @param event [Dry::Event]
+      def on_object_metadata_created(event)
```

Comment:
is `created` a subtype of `updated` for the purpose of indexing?
i think i'd expect the application to publish metadata updated events in both cases, and that the indexer should only need to listen for updates in general, since it's behavior is the same for created cases? thoughts?

Reply:
I did wonder the same, and wanted feedback. I'll refactor the new method out and setup the calls to update.